src/exporter/DaemonMetricCollector.cc

```diff
@@ -41,20 +51,25 @@ void DaemonMetricCollector::request_loop(boost::asio::steady_timer &timer) {
     dump_asok_metrics(sort_metrics, prio_limit, true, dump_response, schema_response, true);
     auto stats_period = g_conf().get_val<int64_t>("exporter_stats_period");
     // time to wait before sending requests again
+    boost::this_thread::interruption_point();
```

Comment:
this can through `boost::thread_interrupted` and I don't see code to handle it

Reply:
Not needed to handle nothing. The only thing needed when reached this line is not continuing repeating the loop used to get the metrics.

When we are terminating Ceph exporter daemon because a SIGINT/SIGTERM signals, we are using this code (in ceph_exporter.cc)

```
  // Stop the server and collector threads by interrupting them
  stop_web_server();
  server_thread.interrupt();  // Interrupt the web server thread
  collector_thread.interrupt();  // Interrupt the DaemonMetricCollector thread  
```

Thanks to the line:

`boost::this_thread::interruption_point();`

When the metrics collector thread reaches this interruption point, it will stop further execution within the loop and can exit cleanly.